### ISSUE TITLE
[BUG] All severity levels of log events are sent to stderr in v1.4.0

### ISSUE THREAD

**Issue by lshi18** [SparebankenVest/azure-key-vault-to-kubernetes#506]
**Note: Make sure to check out known issues (https://akv2k8s.io/troubleshooting/known-issues/) before submitting**

**Components and versions**
Select which component(s) the bug relates to with [X].

[ ] Controller, version: `x.x.x` (docker image tag)
[ x] Env-Injector (webhook), version: `1.4.0` (docker image tag)
[ ] Other

**Describe the bug**
When _log format is configured to `json`_, all severity levels of log events are sent to `stderr`.

**To Reproduce**

```shell
$ docker run -ti --rm \
 --env ENV_INJECTOR_LOG_FORMAT=json \
 --workdir /tmp \
spvest/azure-keyvault-env:1.4.0
```
and in the container run

```shell
/tmp $ /usr/local/bin/azure-keyvault-env >res 2>err
/tmp $ cat res
/tmp $ cat err
{"ts":1679643440208.9385,"caller":"akv2k8s/version.go:31","msg":"version info","v":0,"version":"","commit":"15d87b2","buildDate":"2022-12-08T21:20:01Z","component":"vaultenv"}
{"ts":1679643440209.0002,"caller":"azure-keyvault-env/main.go:184","msg":"azure key vault env injector initializing","v":0}
{"ts":1679643440209.0217,"caller":"azure-keyvault-env/main.go:228","msg":"failed validating config","err":"environment variable ENV_INJECTOR_AUTH_SERVICE_VALIDATION not provided or empty"}
``` 

**Expected behavior**
Up until `v1.3.0`, the behaviour had been to log all severity levels of events to `stdout`, but since `v1.3.1`, the behaviour changes unnoticed to send all levels to the `stderr`.
The expected behaviour is that info-level  logs should be sent to `stdout` instead of `stderr`.

**Logs**

**Additional context**
Repeat the same test in `1.3.0`, all logs are sent to `stdout`.

```shell
$ docker run -ti --rm \
 --env ENV_INJECTOR_LOG_FORMAT=json \
 --workdir /tmp \
spvest/azure-keyvault-env:1.3.0
```
and in the container run

```shell
/tmp $ /usr/local/bin/azure-keyvault-env >res 2>err
/tmp $ cat err
/tmp $ cat res
{"ts":1679643440208.9385,"caller":"akv2k8s/version.go:31","msg":"version info","v":0,"version":"","commit":"15d87b2","buildDate":"2022-12-08T21:20:01Z","component":"vaultenv"}
{"ts":1679643440209.0002,"caller":"azure-keyvault-env/main.go:184","msg":"azure key vault env injector initializing","v":0}
{"ts":1679643440209.0217,"caller":"azure-keyvault-env/main.go:228","msg":"failed validating config","err":"environment variable ENV_INJECTOR_AUTH_SERVICE_VALIDATION not provided or empty"}

**Comment by tspearconquest** [comment]
Some additional context: https://github.com/SparebankenVest/azure-key-vault-to-kubernetes/compare/1.3.0...1.3.1 doesn't appear to show any related commits which would have changed this behavior, however I didn't deeply inspect each one and only looked at the commit titles.

Are you able to still reproduce this with v1.4.0 or higher?

**Comment by 181192** [comment]
A change was done upstream in the k8s logger `klog` for default options. I believe they have corrected it, or at least made in configurable for consumers

**Comment by lshi18** [comment]
Thanks for your reply. As @181192 pointed, since 1.3.1, there are some relavent changes after upgrade the k8s base-component dependency. The upstream now offers [a config called `SplitStream`](https://github.com/kubernetes/component-base/blob/v0.26.5/logs/api/v1/types.go#L74) which is default to false. 

Please note that 

* before 1.3.1,  all logs are written to stdout
* since 1.3.1, all logs are written to stderr

The preferred behavior, I think would be to logging to different streams based on severity. This could be achived by setting the `SplitStream` option here https://github.com/SparebankenVest/azure-key-vault-to-kubernetes/blob/master/cmd/azure-keyvault-env/main.go#L160, i.e., 

```go
		logger, _ := loggerFactory.Create(
			logConfig.LoggingConfiguration{
				Options: logConfig.FormatOptions{
					JSON: logConfig.JSONOptions{SplitStream: true}}})
```

I could prepare an PR, but I think it would be better for you to assess both the feature and technical approach.

**Comment by 181192** [comment]
Fixed in #631 :tada:
